backend/src/apiserver/template/v2_template.go

```diff
@@ -300,9 +311,20 @@ func (t *V2Spec) RunWorkflow(modelRun *model.Run, options RunWorkflowOptions) (u
 		}
 	}
 
+	var pipeline_options *argocompiler.Options
```

Comment:
this looks copied from above. Is there any way to reuse it?

Reply:
It is copied, but the context is different.

First one, `argocompiler.Options` is a non pointer value that represents a copy of the actual data. Second one, `*argocompiler.Options` is a pointer that stores the memory address of the value.

I thought over how this code could be put into a function and called in twice, without having to reuse it. Here's a pseudo code I was thinking might work:

```
func setPipelineOptions(platformSpec map[string]*PlatformSpec, pipelineOptions interface{}) {
	
	for key, platform := range platformSpec {
		if key == "kubernetes" && platform != nil && platform.PipelineConfig != nil {
			if platform.PipelineConfig.SemaphoreKey != "" {
				  SemaphoreKey = platform.PipelineConfig.SemaphoreKey
			}
			if platform.PipelineConfig.MutexName != "" {
			    MutexName = platform.PipelineConfig.MutexName
			}
			break
		}
	}
}
```
and then just calling it each time with the pointer and the non-pointer. But it gets slightly complicated with passing a pointer into a function (needs to be de-referenced).

However, there's a pattern of repetition of the same code between pointer and non-pointer references in this file. For instance [here](https://github.com/kubeflow/pipelines/blob/497bc6b4c5a7f28f158f9616a22420d6982aa04b/backend/src/apiserver/template/v2_template.go#L63-L115), and [here](https://github.com/kubeflow/pipelines/blob/497bc6b4c5a7f28f158f9616a22420d6982aa04b/backend/src/apiserver/template/v2_template.go#L300-L348).

Which is why I would suggest keeping this repetition as is, and we can later refactor this file to remove repeated code by putting them into functions. This could be a separate task out of the scope of this PR/task.

Reply:
Edit: Built a common helper function `getKubernetesSpec` and updated to have the function get called at both instances.